content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md

```diff
@@ -0,0 +1,197 @@
+  to grow the filesystem on that device.
+
+- For various validations at time of node expansion, the CSI driver has to be connected
+  to the backend storage cluster. If the` nodeExpandVolume` request includes a `secretRef`
+  then the CSI driver can make use of the same and connect to the storage cluster to
+  perform the cluster operations.
+
+## How does it work?
+
+To enable this functionality from this version of Kubernetes, Kubernetes Storage SIG have introduced
+a new feature gate called `CSINodeExpandSecret`. Once the feature gate is enabled
+in the cluster, NodeExpandVolume requests can include a `secretRef` field. The NodeExpandVolume request
+is part of CSI; for example, in a request which has been sent from the Kubernetes
+control plane to the CSI driver.
+The admin can specify these secrets as an opaque parameter in storage class as
+they specifies other CSI secrets today.  The secrets mention in the storage class
+has to carry the key name as shown below
+
+```
+csi.storage.k8s.io/node-expand-secret-name: test-secret
+csi.storage.k8s.io/node-expand-secret-namespace: default`
+```
+
+If feature gates are enabled and storage class carries the above secret configuration,
+the subject CSI provisioner receives the credentials from the Secret as part of the NodeExpansion request.
+
+CSI volumes that require secrets for online expansion will have NodeExpandSecretRef
+field set. If not set, the NodeExpandVolume CSI RPC call will be made without a secret.
+
+
+
+## Trying it out
+
+1. Enable the `CSINodeExpandSecret` feature gate (please refer to
+[Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)).
+
+1. Create a Secret, and then a StorageClass that uses that Secret
+
+Here's an example manifest for a Secret that holds credentials:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-secret
+  namespace: default
+data:
+stringData:
+  username: admin
+  password: t0p-Secret
+```
+
+Here's an example manifest for a StorageClass that refers to those credentials:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: csi-hostpath-sc
+parameters:
+  csi.storage.k8s.io/node-expand-secret-name: test-secret   # the name of the Secret
+  csi.storage.k8s.io/node-expand-secret-namespace: default  # the namespace that the Secret is in
+provisioner: hostpath.csi.k8s.io
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+allowVolumeExpansion: true
+```
+
+
+## Example output
+
+If the PersistentVolumeClaim (PVC) was created successfully, you can see that
+configuration within the `spec.csi` field of the PersistentVolume (look for
+`spec.csi.nodeExpandSecretRef`).
+Check that it worked by running `kubectl get persistentvolume <pv_name> -o yaml`.
+You should see something like.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  annotations:
+    pv.kubernetes.io/provisioned-by: hostpath.csi.k8s.io
+  creationTimestamp: "2022-07-23T15:14:07Z"
```

Comment:
(nit)
Change this date to be after the release but before the article publishes.

Reply:
sure put the date after 23rd Aug and before the blog publishing provisioner date of 28 Aug. :+1:

Reply:
Thanks @sftim